src/vs/workbench/services/files/node/fileService.ts

```diff
@@ -366,6 +366,34 @@ export class FileService implements IFileService {
 
 						// 5.) resolve
 						return this.resolve(resource);
+					}, error => {
+						// Can't use 'w' for hidden files, so truncate and use 'r+' if the file exists
+						if (!exists || error.code !== 'EPERM') {
+							throw error;
+						}
+
+						// 5.) truncate
+						return pfs.truncate(absolutePath, 0).then(() => {
+							let writeFilePromise: TPromise<void>;
```

Comment:
@seishun I still suggest to extract those lines which are now duplicate to a separate method that can be called then once from the normal path and once from the error path. that should make the change much smaller and easier to understand 👍